modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -1268,10 +1270,18 @@ protected CompletableFuture<IgniteBiTuple<ClusterNode, Long>> enlist(int partId,
                 throw new TransactionException(REPLICA_UNAVAILABLE_ERR, "Failed to get the primary replica.");
             }
 
+            String consistentId = primaryPeerAndTerm.leader().consistentId();
+
+            ClusterNode node = clusterNodeResolver.apply(consistentId);
+
+            if (node == null) {
+                throw new TransactionException(REPLICA_UNAVAILABLE_ERR, "Failed to resolve the primary replica node, consistentId="
```

Comment:
I would change the error message a bit: `Failed to resolve the primary replica node [consistentId=" + consistentId + ']`